### PR TITLE
CDAP-15428 fixed MD5 hasher plugin to check for field type

### PR DESCRIPTION
**Pull request by albertshau**
Fixed the md5 hasher plugin to validate that the fields specified
are strings and to fail if they are not.

### CODE REVIEW

File: transform-plugins/src/main/java/io/cdap/plugin/Hasher.java [hunk 1]
```diff
@@ -124,14 +131,37 @@ public Config(String hash, String fields) {
...
+                                    "Allowed hashers are md2, md5, sha1, sha256, sha384 and sha512");
       }
+
+      if (inputSchema == null) {
+        return;
+      }
+
+      for (String field : getFields()) {
+        Schema.Field inputField = inputSchema.getField(field);
+        if (inputField == null) {
```
Comment (CuriousVini) [inline transform-plugins/src/main/java/io/cdap/plugin/Hasher.java:151]: Isn't this error case where field to be hashed is not present in input?

File: transform-plugins/src/main/java/io/cdap/plugin/Hasher.java [hunk 2]
```diff
@@ -44,28 +49,27 @@
...
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    StageConfigurer stageConfigurer = pipelineConfigurer.getStageConfigurer();
+    config.validate(stageConfigurer.getInputSchema(), stageConfigurer.getFailureCollector());
+    stageConfigurer.getFailureCollector().getOrThrowException();
+    stageConfigurer.setOutputSchema(stageConfigurer.getInputSchema());
+  }
+
+  @Override
+  public void prepareRun(StageSubmitterContext context) throws Exception {
+    config.validate(context.getInputSchema(), context.getFailureCollector());
```
Comment (CuriousVini) [inline transform-plugins/src/main/java/io/cdap/plugin/Hasher.java]: should throw validation exception using `collector.getOrThrowException()` in prepareRun.

**Review by CuriousVini — COMMENTED** [review]
just 2 comments. rest lgtm

File: transform-plugins/src/main/java/io/cdap/plugin/Hasher.java (hunk 1 above)
Comment (albertshau) [reply transform-plugins/src/main/java/io/cdap/plugin/Hasher.java:151]: The current behavior is that it gets ignored, so I didn't want to change it.

**Review by CuriousVini — APPROVED** [review]

**Comment by albertshau** [comment]
https://builds.cask.co/browse/HYP-BAD588

**Pull request merged**